base/env.jl

```diff
@@ -244,3 +244,30 @@ function withenv(f, keyvals::Pair{T}...) where T<:AbstractString
     end
 end
 withenv(f) = f() # handle empty keyvals case; see #10853
+
+"""
+    username() -> String
+
+Return the username for the current user. If the username cannot be found,
+this function throws an error.
+
+To retrieve a username that is overridable via an environment variable,
+e.g., `USER`, consider using
+```julia
+user = get(ENV, "USER", username())
+```
+
+!!! note
+    The username is determined via the user account and password database
+    that is available only on Unix-like platforms.
```

Comment:
Why?  On Windows we can call [`GetUserNameExA`](https://learn.microsoft.com/en-us/windows/win32/api/secext/nf-secext-getusernameexa) or similar, no?

Doesn't libuv do this for us automatically?

Reply:
According to the [libuv source code](https://github.com/libuv/libuv/blob/da527d8d2a908b824def74382761566371439003/src/win/util.c#L1262), its `getpwuid` function calls `GetUserNameW` on Windows, so it seems like it should work.

Reply:
I saw this https://github.com/JuliaLang/julia/blob/ba984670cf11721928ec8eb528ce2052a2f7b7e2/src/sys.c#L224 and gave up on Windows.

Julia calls `uv_os_get_passwd2` (https://github.com/JuliaLang/julia/blob/master/base/libc.jl#L498) which does not seem to be implemented for Windows (https://github.com/libuv/libuv/blob/da527d8d2a908b824def74382761566371439003/src/win/util.c#L1301)?

Reply:
You could always call the [`GetUserNameW`](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getusernamew) function directly, instead of going through `getuid`.

Something like:
```jl
buf = Vector{Cwchar_t}(undef, 257) # UNLEN+1
len = Ref(UInt32(257))
success = ccall(:GetUserNameW, stdcall, UInt32,
                (Ptr{Cwchar_t}, Ref{UInt32}), buf, len)
windowserror("GetUserNameW", success == 0)
return transcode(String, resize!(buf, len[]))
```

Reply:
Or simply call `uv_os_get_passwd`, rather than `uv_os_get_passwd2`, since it looks like the former *is* supported on Windows.

Reply:
Thank you, Steven. Can you check it now? What do you think about defining the function in `libc.jl`?